File: src/Creator/CreateAdvisor.java
package Creator;

import PersonObject.*;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;


public class CreateAdvisor {
    private ArrayList<Advisor> advisors = new ArrayList<>();
    private String fileName;

    public CreateAdvisor(String fileName,ArrayList<Lecturer> lecturers ) throws JSONException, IOException{
        this.fileName = fileName;
        createAdvisors(lecturers);
    }

    private void createAdvisors(ArrayList<Lecturer> lecturers){
        try {
            String content = new String(Files.readAllBytes(Path.of(fileName)));
            JSONObject jsonObject = new JSONObject(content);
            JSONArray advisorJSON = jsonObject.getJSONArray("advisors");
            for (int i = 0; i < advisorJSON.length(); i++) {
                String name = advisorJSON.getJSONObject(i).getString("name");
                String surname = advisorJSON.getJSONObject(i).getString("surname");
                String advisorId = advisorJSON.getJSONObject(i).getString("advisorId");
                String password = advisorJSON.getJSONObject(i).getString("password");
                Advisor advisor = new Advisor(name, surname, new Id(advisorId), new Password(password));
                advisors.add(advisor);
                lecturers.add(advisor);
            }
        }
        catch (JSONException | IOException ignored){
            System.out.println("An error occurred in the advisors JSON file. Please ensure that the file is created in the correct format and fix any errors.");
            System.exit(0);
        }
    }

    public ArrayList<Advisor> getAdvisors() {
        return advisors;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }
}

File: iter3/python_development/Creator/CreateAdvisor.py
import json
import logging

from iter3.python_development.PersonObject.Advisor import Advisor
from iter3.python_development.PersonObject.Id import Id
from iter3.python_development.PersonObject.Password import Password


class CreateAdvisor:
    def __init__(self, file_name, lecturers):
        self.advisors = []
        self.fileName = file_name
        self.createAdvisors(lecturers)

    def createAdvisors(self, lecturers):
        try:
            with open(self.fileName, 'r') as file:
                content = file.read()
                json_object = json.loads(content)
                advisor_json = json_object['advisors']
                for advisor_data in advisor_json:
                    name = advisor_data['name']
                    surname = advisor_data['surname']
                    advisor_id = advisor_data['advisorId']
                    password = advisor_data['password']
                    advisor = Advisor(name, surname, Id(advisor_id), Password(password))
                    self.advisors.append(advisor)
                    lecturers.append(advisor)
        except (json.JSONDecodeError, FileNotFoundError, IOError) as e:
            logging.exception(f"Error in advisor date in advisors.json file: {e}")
            exit(0)

    def getAdvisors(self):
        return self.advisors

    def getFileName(self):
        return self.fileName

    def setFileName(self, file_name):
        self.fileName = file_name
